src/containers/pages/OrganizationViews/AssignPractitioners/index.tsx

```diff
@@ -0,0 +1,279 @@
+interface AssignPractitionerProps {
+  fetchOrganizationsCreator: typeof fetchOrganizations;
+  fetchPractitionerRolesCreator: typeof fetchPractitionerRoles;
+  organization: Organization | null;
+  serviceClass: typeof OpenSRPService;
+  assignedPractitioners: Practitioner[];
+}
+
+/** default props for AssignPractitioner component */
+const defaultAssignPractitionerProps: AssignPractitionerProps = {
+  assignedPractitioners: [],
+  fetchOrganizationsCreator: fetchOrganizations,
+  fetchPractitionerRolesCreator: fetchPractitionerRoles,
+  organization: null,
+  serviceClass: OpenSRPService,
+};
+
+/** type intersection for all types that pertain to the props */
+export type PropsTypes = AssignPractitionerProps & RouteComponentProps<RouteParams>;
+
+/** AssignPractitioner component */
+const AssignPractitioner: React.FC<PropsTypes> = props => {
+  const {
+    serviceClass,
+    fetchOrganizationsCreator,
+    organization,
+    fetchPractitionerRolesCreator,
+    assignedPractitioners,
+  } = props;
+  const [selectedOptions, setSelectedOptions] = useState<OptionsType<SelectOption>>([]);
+
+  useEffect(() => {
+    const organizationId = props.match.params.id;
+    loadOrganization(organizationId, serviceClass, fetchOrganizationsCreator);
+    loadOrgPractitioners(organizationId, serviceClass, fetchPractitionerRolesCreator);
+  }, []);
+
+  if (!organization) {
+    return <Loading />;
+  }
+
+  // TODO - scenario where this request doesn't return all practitioners in a single query
+  /** load all practitioners at least all of those returned
+   * in a single call to Practitioners endpoint
+   * @param {typeof OpenSRPService} service - the OpenSRPService
+   */
+  const loadAllPractitioners = async (service: typeof OpenSRPService = OpenSRPService) => {
+    const serve = new service(OPENSRP_PRACTITIONER_ENDPOINT);
+    const allPractitioners = await serve.list();
+    return formatOptions(allPractitioners);
+  };
+
+  // TODO - Hack in this: typings for the changeHandler function.
+  /** This sets the state selectedOptions
+   * @param {ValueType<SelectOption>} -  the so far selected options
+   * @param {ActionMeta} - information on the change event; custom react-select event
+   */
+  const changeHandler = (chosenOptions: any, { action, removedValue, option }: any) => {
+    if (!chosenOptions) {
+      return;
+    }
+    switch (action) {
+      case 'remove-value':
+      case 'pop-value':
+        if (removedValue.isFixed) {
+          return;
+        } else {
+          const remainingOptions = selectedOptions.filter(o => o.value !== removedValue.value);
+          setSelectedOptions(remainingOptions);
+        }
+        break;
+      case 'select-option':
+        setSelectedOptions([...selectedOptions, option]);
+        break;
+      case 'clear':
+        chosenOptions = chosenOptions.filter((v: any) => !v.isFixed);
+    }
+  };
+
+  // TODO - This will initiate an api request for the same exact data each time someone types
```

Comment:
What do you think needs to be done to resolve this TODO - should we create an issue with those details?